server/channels/app/mention_keywords.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+const (
+	mentionableUserPrefix  = "user:"
+	mentionableGroupPrefix = "group:"
+)
+
+type MentionableID string
+
+func MentionableUserID(userID string) MentionableID {
+	return MentionableID(fmt.Sprint(mentionableUserPrefix, userID))
+}
+
+func MentionableGroupID(groupID string) MentionableID {
+	return MentionableID(fmt.Sprint(mentionableGroupPrefix, groupID))
+}
+
+func (id MentionableID) AsUserID() (userID string, ok bool) {
+	idString := string(id)
+	if !strings.HasPrefix(idString, mentionableUserPrefix) {
+		return "", false
+	}
+
+	return idString[len(mentionableUserPrefix):], true
+}
```

Comment:
Would it make sense for `MentionableID` to be a `struct` identifying both the `id` and its type? (Then we could just `switch` directly on `mentionableID.Type` where needed, potentially simplifying consumers.)

Reply:
I feel like I had a reason to not use a struct, but I can't remember why now. I think I may have started with just trying to use strings before realizing that a named type would help me find everywhere these were used.

Unless anyone has strong objections, I'm going to leave it as-is. I'm not too worried about simplifying consumers since I'm only actually using `AsUserID` and `AsGroupID` in one place at the moment. It could arguably be made unexported although I think that might conflict with variable names in a few places

Reply:
So long as it remains an "internal" implementation detail, I'm 0/5. The only thought might be the overhead of an unnecessary string allocation to apply the prefix, but I'm even -1/5 about worrying about that for now. Will leave to you!